common/src/main/java/dev/itsmeow/quickhomes/QuickHomesMod.java

```diff
@@ -33,8 +34,18 @@ public static void registerCommands(CommandDispatcher dispatcher) {
         dispatcher.register(Commands.literal("home").requires(isPlayer).executes(command -> {
             ServerPlayer player = command.getSource().getPlayerOrException();
             Pair<Vec3, ResourceKey<Level>> home = ((IStoreHome) player).getHome();
-            if(home.getLeft() != null && home.getRight() != null) {
-                player.teleportTo(player.getServer().getLevel(home.getRight()), home.getLeft().x, home.getLeft().y, home.getLeft().z, player.getYRot(), player.getXRot());
+            Vec3 homePos = home.getLeft();
+            ResourceKey<Level> homeLevel = home.getRight();
+            if(homePos != null && homeLevel != null) {
+                if (player.isPassenger()) {
+                    Entity vehicle = player.getVehicle();
+                    // If player is not riding with or on another player, teleport the vehicle along with passengers.
+                    if (!vehicle.getPassengersAndSelf().anyMatch(entity -> (!entity.equals(player)) && entity instanceof Player)) {
+                        vehicle.teleportTo(homePos.x, homePos.y, homePos.z);
```

Comment:
This doesn't handle the vehicle changing dimensions

Reply:
Yep, completely forgot about that part. Take a look now!